Makefile

```diff
@@ -427,20 +427,39 @@ __check_defined = \
 	$(if $(value $1),, \
 		$(error Undefined environment variable $1$(if $2, ($2))))
 
-generate-hypershift-deplyment:
-	@:$(call check_defined, MGMT_AWS_CONFIG_PATH)
-	@:$(call check_defined, TMPDIR)
+# yq: https://github.com/mikefarah/yq
+yq ?= yq
 
-	mkdir -p ${TMPDIR}/hypershift-cli
-	oc image extract registry.ci.openshift.org/ci/hypershift-cli:latest --path /usr/bin/hypershift:${TMPDIR}/hypershift-cli --confirm
-	chmod +x ${TMPDIR}/hypershift-cli/hypershift
+generate-hypershift-deployment:
+	@:$(call check_defined, MGMT_AWS_CONFIG_PATH)
 
-	${TMPDIR}/hypershift-cli/hypershift \
+	$(SKIP_PULL) || $(CONTAINER_ENGINE) pull --platform linux/${GO_ARCH} registry.ci.openshift.org/ci/hypershift-cli:latest
+	$(CONTAINER_ENGINE) run $(USER) --platform linux/${GO_ARCH} \
+		--rm \
+		-v "$(MGMT_AWS_CONFIG_PATH):/mgmt-aws$(VOLUME_MOUNT_FLAGS)" \
+		registry.ci.openshift.org/ci/hypershift-cli:latest \
 		install \
 		--oidc-storage-provider-s3-bucket-name=hypershift-oidc-provider \
-		--oidc-storage-provider-s3-credentials=${MGMT_AWS_CONFIG_PATH} \
+		--oidc-storage-provider-s3-credentials=/mgmt-aws \
 		--oidc-storage-provider-s3-region=us-east-1 \
 		--hypershift-image=registry.ci.openshift.org/ci/hypershift-cli:latest \
 		--enable-uwm-telemetry-remote-write=false \
-		render > $(CURDIR)/clusters/hive/hypershift/hypershift-install.yaml
-.PHONY: generate-hypershift-deplyment
+		render | $(yq) eval 'select(.kind != "Secret")' > clusters/hive/hypershift/hypershift-install.yaml
+
+	# The following lines will avoid using docker image
```

Comment:
Let us remove those unused cmds or use them with if-else.

Reply:
I do not feel this is the right place to keep things.
You can save it in your own fork in case you need it in the future.